src/main/java/org/folio/oaipmh/processors/MarcWithHoldingsRequestHelper.java

```diff
@@ -719,12 +720,16 @@ private HttpClientRequest createEnrichInventoryClientRequest(HttpClient httpClie
   private void doPostRequestToSrs(SourceStorageSourceRecordsClient srsClient, Vertx vertx, boolean deletedRecordSupport,
       List<String> listOfIds, AtomicInteger attemptsCount, int retryAttempts, Promise<Map<String, JsonObject>> promise) {
     try {
+      AtomicBoolean retryTriggered = new AtomicBoolean();
       srsClient.postSourceStorageSourceRecords("INSTANCE", deletedRecordSupport, listOfIds, srsResponse -> {
         int statusCode = srsResponse.statusCode();
         String statusMessage = srsResponse.statusMessage();
         srsResponse.exceptionHandler(e -> {
-          logger.error("Error has been occurred while requesting SRS: " + e.getMessage(), e);
-          retrySRSRequest(srsClient, vertx, deletedRecordSupport, listOfIds, attemptsCount, retryAttempts, promise, statusCode, statusMessage);
+          logger.error("Error has been occurred while requesting SRS kek1: " + e.getMessage(), e);
+          if (!retryTriggered.get()) {
```

Comment:
Do it atimically, otherwise you have race condition

Reply:
Thanks, done